pycbc/waveform/bank.py

```diff
@@ -679,7 +686,7 @@ def generate_with_delta_f_and_max_freq(self, t_num, max_freq, delta_f,
         if cached_mem is None:
             wav_len = int(max_freq / delta_f) + 1
             cached_mem = zeros(wav_len, dtype=np.complex64)
-        if self.compressed_waveforms is not None :
+        if self.check_compressed_waveforms:
```

Comment:
Ok, so in `FilterBank.__init__`, replace `load_compressed` and `load_compressed_now` with `use_compressed=True`. Save that as a class attribute. Then, down here, do `if self.has_compressed_waveforms and self.use_compressed:`.